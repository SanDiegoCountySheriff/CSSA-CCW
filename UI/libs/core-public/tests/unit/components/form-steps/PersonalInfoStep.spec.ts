// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck
import PersonalInfoStep from '@core-public/components/form-stepper/form-steps/PersonalInfoStep.vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';

const pinia = createTestingPinia();
const localVue = createLocalVue();
const tMock = {
  $t: text => text,
};

describe('PersonalInfoStep', () => {
  let vuetify;
  let wrapper;

  beforeEach(() => {
    vuetify = new Vuetify();
    wrapper = mount(PersonalInfoStep, {
      localVue,
      vuetify,
      pinia,
      mocks: tMock,
      propsData: {
        handleNextSection: () => null,
      },
      data() {
        return {
          valid: false,
          personalInfo: {},
          aliases: [],
          errors: [],
          ssnConfirm: '',
        };
      },
    });
  });

  it('Should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Should render the correct amount of inputs', () => {
    const textFields = wrapper.findAllComponents('.v-text-field');

    expect(textFields).toHaveLength(7);
  });

  it('Should update the state on input change', async () => {
    const lastNameField = wrapper.find('#last-name-field');

    await lastNameField.setValue('last');
    expect(lastNameField.element.value).toBe('last');
  });
});
