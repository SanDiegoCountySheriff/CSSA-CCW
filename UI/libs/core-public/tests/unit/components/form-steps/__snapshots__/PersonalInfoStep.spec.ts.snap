// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalInfoStep Should match the snapshot 1`] = `
<div>
  <form novalidate="novalidate" class="v-form">
    <div class="v-subheader subHeader font-weight-bold theme--light">
      Personal Information
    </div>
    <div class="row ml-5">
      <div class="col-lg-6 col-12">
        <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-text-field__slot"><label for="last-name-field" class="v-label theme--light" style="left: 0px; position: absolute;">Last name</label><input maxlength="50" aria-haspopup="true" aria-expanded="false" id="last-name-field" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
              <div class="v-counter theme--light">0 / 50</div>
            </div>
          </div>
        </div><span class="v-tooltip v-tooltip--bottom"><!----></span>
      </div>
      <div class="col-lg-6 col-12">
        <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-text-field__slot"><label for="input-10" class="v-label theme--light" style="left: 0px; position: absolute;">First name</label><input maxlength="50" id="input-10" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
              <div class="v-counter theme--light">0 / 50</div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-lg-6 col-12">
        <div class="v-input pl-6 v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
          <div class="v-input__control">
            <div class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-text-field__slot"><label for="input-14" class="v-label theme--light" style="left: 0px; position: absolute;">Middle name</label><input maxlength="50" id="input-14" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
              <div class="v-counter theme--light">0 / 50</div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-lg-6 col-12">
        <div class="v-input pl-6 v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
          <div class="v-input__control">
            <div class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-text-field__slot"><label for="input-17" class="v-label theme--light" style="left: 0px; position: absolute;">Maiden name</label><input maxlength="50" id="input-17" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
              <div class="v-counter theme--light">0 / 50</div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-lg-6 col-12">
        <div class="v-input pl-6 v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
          <div class="v-input__control">
            <div class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-text-field__slot"><label for="input-20" class="v-label theme--light" style="left: 0px; position: absolute;">Suffix</label><input maxlength="10" id="input-20" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
              <div class="v-counter theme--light">0 / 10</div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr role="separator" aria-orientation="horizontal" class="my-3 v-divider theme--light">
    <div class="v-subheader sub-header font-weight-bold theme--light">
      Social Security Information
    </div>
    <div class="row ml-5">
      <div class="col-lg-6 col-12">
        <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-text-field__slot"><label for="input-25" class="v-label theme--light" style="left: 0px; position: absolute;">Social Security Number</label><input id="input-25" type="password"></div>
              <div class="v-input__append-inner">
                <div class="v-input__icon v-input__icon--append"><button type="button" aria-label="append icon" class="v-icon notranslate v-icon--link mdi mdi-eye-off theme--light"></button></div>
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-lg-6 col-12">
        <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-text-field__slot"><label for="input-30" class="v-label theme--light" style="left: 0px; position: absolute;">Confirm SSN</label><input id="input-30" type="password"></div>
              <div class="v-input__append-inner">
                <div class="v-input__icon v-input__icon--append"><button type="button" aria-label="append icon" class="v-icon notranslate v-icon--link mdi mdi-eye-off theme--light"></button></div>
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr role="separator" aria-orientation="horizontal" class="my-3 v-divider theme--light">
    <div class="v-subheader subHeader font-weight-bold theme--light">
      Marital Status
    </div>
    <div class="row ml-1">
      <div class="col-lg-6 col-12">
        <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-select">
          <div class="v-input__control">
            <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-37" class="v-input__slot">
              <fieldset aria-hidden="true">
                <legend style="width: 0px;"><span class="notranslate">​</span></legend>
              </fieldset>
              <div class="v-select__slot"><label for="input-37" class="v-label theme--light" style="left: 0px; position: absolute;">Marital status</label>
                <div class="v-select__selections"><input id="input-37" readonly="readonly" type="text" aria-readonly="false" autocomplete="off"></div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                </div><input type="hidden">
              </div>
              <div class="v-menu">
                <!---->
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
      <!---->
    </div>
  </form>
  <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
  <div class="v-subheader sub-header font-weight-bold theme--light">
    Aliases
  </div>
  <div class="v-input ml-6 theme--light v-input--selection-controls v-input--radio-group v-input--radio-group--row">
    <div class="v-input__control">
      <div class="v-input__slot" style="height: auto;">
        <div role="radiogroup" aria-labelledby="input-44" class="v-input--radio-group__input">
          <legend id="input-44" class="v-label theme--light" style="left: 0px; position: relative;">In the past have you ever gone by a different name?</legend>
          <div class="v-radio ml-6 theme--light">
            <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-45" role="radio" type="radio" name="radio-44" value="true">
              <div class="v-input--selection-controls__ripple"></div>
            </div><label for="input-45" class="v-label theme--light" style="left: 0px; position: relative;">Yes</label>
          </div>
          <div class="v-radio ml-6 theme--light v-item--active">
            <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-marked theme--light primary--text"></i><input aria-checked="true" id="input-47" role="radio" type="radio" name="radio-44" value="false">
              <div class="v-input--selection-controls__ripple primary--text"></div>
            </div><label for="input-47" class="v-label theme--light" style="left: 0px; position: relative;">No</label>
          </div>
        </div>
      </div>
      <div class="v-messages theme--light">
        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
      </div>
    </div>
  </div>
  <!---->
  <hr role="separator" aria-orientation="horizontal" class="my-5 v-divider theme--light">
  <div class="form-btn-container">
    <div class="form-btn-inner">
      <div><button type="button" disabled="disabled" class="mt-3 v-btn v-btn--disabled v-btn--has-bg theme--light v-size--small"><span class="v-btn__content">
        Continue
      </span></button> <button type="button" disabled="disabled" class="mt-3 v-btn v-btn--disabled v-btn--has-bg theme--light v-size--small"><span class="v-btn__content">
        Save and Exit
      </span></button></div>
      <div><button type="button" class="mt-3 v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small warning mx-2"><span class="v-btn__content">
        Go back
      </span></button> <button type="button" class="mt-3 v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small error mr-2"><span class="v-btn__content">
        Cancel
      </span></button></div>
    </div>
  </div>
  <!---->
  <div class="v-snack v-snack--bottom">
    <transition-stub name="v-snack-transition">
      <div class="v-snack__wrapper v-sheet v-sheet--outlined theme--light error--text" style="display: none;">
        <div role="status" aria-live="polite" class="v-snack__content">
          Section update unsuccessful please try again.
        </div>
        <div class="v-snack__action "></div>
      </div>
    </transition-stub>
  </div>
</div>
`;
